src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -374,7 +380,7 @@ - (void)centralManager:(CBCentralManager *)central
     }
 
     const uint8_t * bytes = (const uint8_t *) [serviceData bytes];
-    if ([serviceData length] != 8) {
+    if ([serviceData length] != sizeof(ChipBLEDeviceIdentificationInfo)) {
```

Comment:
This would make more sense if we then copied the data into a ChipBLEDeviceIdentificationInfo.  Which would let us take out the manual opcode/discriminator parsing.  Followup for that, please.

Reply:
Yeah we do that in the other places... not sure why this one function does its own parsing.